src/hooks/useCheckImage.ts

```diff
@@ -1,22 +1,41 @@
+import { useMemo } from "react";
 import { useAsync } from "react-use";
 
 export const useCheckImage = (
-  src: string
+  src: string | undefined
```

Comment:
maybe better would be this way? or would you prefer it to be more explicit?
```suggestion
  src?: string
```

Reply:
This a not a completely equal substitution. With `src?: string` one can not pass an argument to `useCheckImage` at all while with `string | undefined` the argument must be passed but can be undefined. My intention is to indicate that `useCheckImage` needs an argument but it can be of unspecified value.